src/main/kotlin/eda/shoppingBasket/service/entity/ShoppingBasket.kt

```diff
@@ -0,0 +1,22 @@
+package eda.shoppingBasket.service.entity
+
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.OneToMany
+import java.util.UUID
+
+@Entity
+class ShoppingBasket(
+    val customerID: UUID,
+    var totalPrice: Float = 0.0f,
+    @Id
+    val shoppingBasketID: UUID = UUID.randomUUID()
+) {
+    @OneToMany
```

Comment:
If you want to have bidirectional references, you'll have to justify that probably (Because they are more error prone). If you switch to unidirectional, you'd have to do more work in the service with a method in the ShoppingItemRepo `findByShoppingBasket` / `findByShoppingBasketId`. Also, if we have unidirectional, we should go "from specific to general" So from the small parts upward

Reply:
https://www.baeldung.com/jpa-hibernate-associations

Reply:
i decided on a bidirectional reference as it provides more flexibility and i thought we might need to navigate from one repo to another quite often. since we do not have a lot of complex operations (imo), maybe we can just keep using bidirectional just coz it is more convenient? 
if it actually shows performance (and/or other) issues, we can always switch to unidirectional (or is it not as simple as i think it is?)

Reply:
It's easy to switch (at the moment) and it's more convenient, yes. BUT: Bidircetional references create cyclic dependencies. If we later wanted to use the ShoppingBasketItem in a different context (maybe analasys or sth) We would load the shopping basket each time, even though we dont need it. Even if we only access one specific item (to see which customers bought it for example, we would first load the item, the shopping basket, and by that all of the other items in the basket. 
As for the direction, iirc there was some pattern that stated you should go from specific to general (keep the reference in the more specific class, as you would end up with a lot of references in the general class, when every addition is kept in it. 

Don't know if that was easy to read, if you want we can chat about it via discord, maybe in a call? I'd need to look up some more sources too.